tests/formatter.rs

```diff
@@ -795,8 +801,9 @@ bar = { version = "0.1.0", optional = true }
 error: unused optional dependency
   |
 4 |   bar = { version = "0.1.0", optional = true }
-  |  __________________________________________^
-  |                              --------------- info: This should also be long but not too long
+  |  ____________________________--------------^
+  | |                            |
+  | |                            info: This should also be long but not too long
```

Comment:
Am I reading this right that the annotation spans overlap but the rendering is not showing that?

Also, whats with the trailing caret?

Reply:
Yes, they do overlap, but it is being shown. Multiline annotation underlines get written first, then any `-` and `^`. In this case it is showing they overlap with the `-` intersecting the multiline annotation.